File: 0792-binary-search/0792-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        x=self.binary_search(nums,0,len(nums)-1,target)
        return x

    def binary_search(self,nums,low,high,target):
        mid= (low+high)//2
        if low>high:
            return -1
        if target == nums[mid]:
            return mid
        elif target < nums[mid]:
            return self.binary_search(nums,low,mid-1,target)
        else:
            return self.binary_search(nums,mid+1,high,target) 
       



        

        

File: 0792-binary-search/0792-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int low=0;
        int high= nums.length-1;
        return binarySearch(nums,target,low,high);
        
    }
    public int binarySearch(int[] nums, int target,int low,int high){

            if(low>high){
                return -1;

            }
            int mid= (low+high)/2;
            if(nums[mid]==target){
                return mid;
            }
            else if(target>nums[mid]){
                return binarySearch(nums,target,mid+1,high);
            }
            else{
                return binarySearch(nums,target,low,mid-1);
            }
    }
}